src/config/language.c

```diff
@@ -53,6 +74,33 @@ static struct str_map {
     u16 pos;
 } lookupmap[MAX_STRINGS];
 
+static const char* LoadString(u16 pos, const char *str)
+{
+#if !SUPPORT_DYNAMIC_LOCSTR
+    (void)str;
+    return strings + pos;
+#else
+    char *ret = &strcache[str_ptr];
+    char buf[MAX_LINE];
+    fseek(fh, pos, SEEK_SET);
+    if (fgets(buf, MAX_LINE, fh) == NULL)
+        return str;
+
+    fix_crlf(buf);
+    unsigned len = strlen(buf);
+    if (len + str_ptr >= MAX_STRING_BUFFER - 1)
+    {
```

Comment:
I don't understand this logic.  What happens in the case that the cache is full?

Reply:
start from 0 again if cache is full. since the cache only for the case like snprintf("". _tr(a), _tr(b)). I assume we will not use many _tr calls and cache is always long enough. We don't have something to give us clear signal we don't need the last result of _tr already. the cases for this type of usage is not many. But cleanup them take a bit more effort.

Reply:
maybe we can reset cache after snprintf/sprintf returns. but I don't like this approach as it connected two indendent things together.